scripts/production/homology_stats_rapid.py

```diff
@@ -0,0 +1,189 @@
+# regarding copyright ownership.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This script fetches the number of coding genes and number of coding genes with homologies
+against the taxonomically closest reference species.
+
+Usage example::
+    python homology_stats_rapid.py\
+    -d mysql://ensro@mysql-ens-sta-5:4684/accipiter_gentilis_compara_109\
+    -r ensembl_compara_references -o gallus.json
+"""
+
+import argparse
+import json
+from typing import Tuple, List
+
+from sqlalchemy import create_engine, text
+from sqlalchemy.engine import Connection
+from ensembl.database import DBConnection
+from ensembl.ncbi_taxonomy.api.utils import Taxonomy
+
+
+def parse_arguments() -> argparse.Namespace:
+    """
+    Parse command line arguments.
+
+    Returns:
+        argparse.Namespace: Parsed command line arguments
+    """
+    parser = argparse.ArgumentParser(
+        description="Query gene and homology counts from an Ensembl compara rapid release database."
+    )
+    parser.add_argument("-d", "--database", required=True, help="Database URL")
+    parser.add_argument(
+        "-r",
+        "--ref_db",
+        required=True,
+        help="Reference database name. Must be on same server as target database.",
+    )
+    parser.add_argument(
+        "-o", "--output", required=True, help="Output file in JSON format"
+    )
+    return parser.parse_args()
+
+
+def get_closest_ref(
+    tax_dbc: DBConnection, rr_dbc: Connection, ref_db: str
+) -> Tuple[int, str, int, str, int]:
+    """
+    Get the closest reference for the given rapid release database.
+
+    Args:
+        tax_dbc (DBConnection): Taxonomy database connection
+        rr_dbc (Connection): Rapid release database connection
+        ref_db (str): Reference database name
+
+    Returns:
+        Tuple[int, str, int, str, int]: Result tuple
+    """
+    query1 = """
+    SELECT name, taxon_id FROM genome_db LIMIT 1;
+    """
+
+    with rr_dbc.connect() as conn:
+        tmp = conn.execute(text(query1)).one()
+        query_species, query_taxid = tmp["name"], tmp["taxon_id"]
+
+    query2 = f"""
+    SELECT genome_db_id, gdb.taxon_id, gdb.name FROM
+    species_set JOIN {ref_db}.genome_db gdb USING(genome_db_id)
+    WHERE genome_db_id > 100;
+    """
+
+    with rr_dbc.connect() as conn, tax_dbc.session_scope() as session:
+        result = conn.execute(text(query2))
+        res: List[Tuple] = []
+        for dict_row in result.mappings():
```

Comment:
I'd suggest to consider accessing the result as a tuple instead of using `mappings`, particularly if you use a single `DBConnection` object.
```suggestion
        for genome_db_id, taxid, name in result:
```